### PR TITLE
Handle offset for menu bar only when on top; error handing and read on `-m config position`

### PR DESCRIPTION
**Pull request by donaldguy**
It seems to work fine with the drawing code and check thats there:

https://github.com/FelixKratz/SketchyBar/blob/5ba9ad9e0f7b4a4bce65a23a2f3a431d1d16e2d0/src/bar.c#L278-L282

CleanShot 2021-08-18 at 17 41 16@2x

### CODE REVIEW

File: src/message.c [hunk 1]
```diff
@@ -367,6 +367,17 @@ static void handle_domain_config(FILE *rsp, struct token domain, char *message)
     } else {
       daemon_fail(rsp, "value for '%.*s' must be either 'main' or 'all'.\n", command.length, command.text);
     }
+  } else if (token_equals(command, COMMAND_CONFIG_BAR_POSITION)) {
+    int length = strlen(message);
+    char * top = "top";
+    char * bottom = "bottom";
```
Comment (donaldguy) [inline src/message.c:373]: I'm [not personally a fan of / don't understand any advantages to] this pre-decleration of once-used string literals, but I just tried to follow your style as it is

File: src/message.c [hunk 2]
```diff
@@ -367,6 +367,17 @@ static void handle_domain_config(FILE *rsp, struct token domain, char *message)
...
+    if (length <= 0) {
+      fprintf(rsp, "%s\n", g_bar_manager.position);
+    } else if ((strcmp(message,top) == 0) || (strcmp(message,bottom) == 0)) {
+      bar_manager_set_position(&g_bar_manager, string_copy(message));
+    } else {
+      daemon_fail(rsp, "value for '%.*s' must be either 'main' or 'all'.\n", command.length, command.text);
```
Comment (donaldguy) [inline src/message.c:379]:
```suggestion
      daemon_fail(rsp, "value for '%.*s' must be either 'top' or 'bottom'.\n", command.length, command.text);
```

**Comment by donaldguy** [comment]
... It seems to break click handling though  ...

(I'd convert this PR to a draft but that feature seems to be disabled on the repo?)

**Comment by FelixKratz** [comment]
Will fix clicking now... just a sec

**Comment by FelixKratz** [comment]
Should be good now

File: src/message.c (hunk 1 above)
Comment (FelixKratz) [reply src/message.c:373]: Yeah me neither, its fixed on master.

**Comment by donaldguy** [comment]
I was getting there :-) but was held up having feelings about less directly important things.

Anyway - gonna go ahead and offer those, add back the error/fetch handling (though ... something, something token system):

**Review by FelixKratz — APPROVED** [review]
Yes, looks great.

**Pull request merged**